crates/shared/src/price_estimation/trade_verifier.rs

```diff
@@ -106,13 +105,19 @@ impl TradeVerifier {
         trade: &TradeKind,
         out_amount: &U256,
     ) -> Result<Estimate, Error> {
-        if verification.from.is_zero() {
-            // Don't waste time on common simulations which will always fail.
-            return Err(anyhow::anyhow!("trader is zero address").into());
-        }
-
         let start = std::time::Instant::now();
 
+        // prepare state overrides which may alter the trader
+        // address to enable more verifications
+        let (overrides, verification) = {
+            let mut verification = verification.clone();
```

Comment:
Now `verify_inner()` function requires `verification` to be mutable so I think changing `verify_inner()` argument to `verification: &mut Verification` will be more cleaner here as we don't need to create the `(overrides, verification)` tuple anymore, but just call `prepare_state_overrides()` function. Whereas `verification.clone()` can be moved to the caller function: `inner()` directly to the `verify_inner()` call argument.